IllegalAccessException during mod loading process

### Describe the bug

I don't know how to describe it, but IllegalAccessExcetpion
Log provided:
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:32)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.Launcher.run(Launcher.java:108)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.Launcher.main(Launcher.java:78)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@1.1.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at MC-BOOTSTRAP/fmlloader@1.20.1-47.1.3/net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111)
	at MC-BOOTSTRAP/fmlloader@1.20.1-47.1.3/net.minecraftforge.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:99)
	at MC-BOOTSTRAP/fmlloader@1.20.1-47.1.3/net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$makeService$0(CommonClientLaunchHandler.java:25)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30)
	... 7 more
Caused by: java.lang.RuntimeException: java.lang.IllegalAccessError: class appeng.block.misc.MysteriousCubeBlock tried to access private method 'boolean net.minecraft.world.level.block.Blocks.m_50805_(net.minecraft.world.level.block.state.BlockState, net.minecraft.world.level.BlockGetter, net.minecraft.core.BlockPos)' (appeng.block.misc.MysteriousCubeBlock is in module ae2@15.0.6-beta of loader 'TRANSFORMER' @834e986; net.minecraft.world.level.block.Blocks is in module minecraft@1.20.1 of loader 'TRANSFORMER' @834e986)
	at MC-BOOTSTRAP/fmlloader@1.20.1-47.1.3/net.minecraftforge.fml.loading.BackgroundWaiter.runAndTick(BackgroundWaiter.java:32)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.client.main.Main.main(Main.java:151)
	... 15 more
Caused by: java.lang.IllegalAccessError: class appeng.block.misc.MysteriousCubeBlock tried to access private method 'boolean net.minecraft.world.level.block.Blocks.m_50805_(net.minecraft.world.level.block.state.BlockState, net.minecraft.world.level.BlockGetter, net.minecraft.core.BlockPos)' (appeng.block.misc.MysteriousCubeBlock is in module ae2@15.0.6-beta of loader 'TRANSFORMER' @834e986; net.minecraft.world.level.block.Blocks is in module minecraft@1.20.1 of loader 'TRANSFORMER' @834e986)
	at TRANSFORMER/ae2@15.0.6-beta/appeng.block.misc.MysteriousCubeBlock.<clinit>(MysteriousCubeBlock.java:24)
	at TRANSFORMER/ae2@15.0.6-beta/appeng.core.definitions.AEBlocks.block(AEBlocks.java:273)
	at TRANSFORMER/ae2@15.0.6-beta/appeng.core.definitions.AEBlocks.block(AEBlocks.java:263)
	at TRANSFORMER/ae2@15.0.6-beta/appeng.core.definitions.AEBlocks.<clinit>(AEBlocks.java:157)
	at TRANSFORMER/ae2@15.0.6-beta/appeng.core.AppEngBootstrap.runEarlyStartup(AppEngBootstrap.java:73)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.Bootstrap.handler$zzd000$initRegistries(Bootstrap.java:541)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.Bootstrap.m_135870_(Bootstrap.java:57)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.client.main.Main.lambda$main$0(Main.java:151)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

### How to reproduce the bug

Grab the latest forge release and attempt to launch the game

### Expected behavior

Game launches with no errors

### Additional details

Did not occur on mod version 15.0.5-beta for Forge

### Which minecraft version are you using?

1.20

### On which mod loaders does it happen?

Forge

### Crash log

https://pastebin.com/srUWuPyy

--- a/src/main/java/appeng/mixins/EarlyStartupMixin.java
+++ b/src/main/java/appeng/mixins/EarlyStartupMixin.java
@@ -24,6 +24,7 @@ import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 import net.minecraft.server.Bootstrap;
+import net.minecraftforge.fml.ModLoader;
 
 import appeng.core.AppEngBootstrap;
 
@@ -38,6 +39,10 @@ public abstract class EarlyStartupMixin {
     // and we don't want this to be invoked from the nested call since MC isn't fully initialized by then.
     @Inject(at = @At(value = "INVOKE", target = "net/minecraftforge/registries/GameData. vanillaSnapshot()V", shift = At.Shift.AFTER, by = 1), method = "bootStrap")
     private static void initRegistries(CallbackInfo ci) {
+        if (!ModLoader.isLoadingStateValid()) {
+            return;
+        }
+
         AppEngBootstrap.runEarlyStartup();
     }
 

